Userland/Libraries/LibJS/Tests/builtins/Temporal/ZonedDateTime/ZonedDateTime.prototype.round.js

```diff
@@ -0,0 +1,108 @@
+describe("correct behavior", () => {
+    test("length is 1", () => {
+        expect(Temporal.ZonedDateTime.prototype.round).toHaveLength(1);
+    });
+
+    test("basic functionality", () => {
+        const zonedDateTime = new Temporal.ZonedDateTime(
+            1111111111111n,
+            new Temporal.TimeZone("UTC")
+        );
+        expect(zonedDateTime.round({ smallestUnit: "second" }).epochNanoseconds).toBe(
+            1111000000000n
+        );
+        expect(
+            zonedDateTime.round({ smallestUnit: "second", roundingMode: "ceil" }).epochNanoseconds
+        ).toBe(1112000000000n);
+        expect(
+            zonedDateTime.round({
+                smallestUnit: "minute",
+                roundingIncrement: 30,
+                roundingMode: "floor",
+            }).epochNanoseconds
+        ).toBe(0n);
+        expect(
+            zonedDateTime.round({
+                smallestUnit: "minute",
+                roundingIncrement: 30,
+                roundingMode: "halfExpand",
+            }).epochNanoseconds
+        ).toBe(1800000000000n);
+    });
+});
+
+describe("errors", () => {
+    test("this value must be a Temporal.ZonedDateTime object", () => {
+        expect(() => {
+            Temporal.ZonedDateTime.prototype.round.call("foo");
+        }).toThrowWithMessage(TypeError, "Not an object of type Temporal.ZonedDateTime");
+    });
+
+    test("missing options object", () => {
+        expect(() => {
+            const zonedDateTime = new Temporal.ZonedDateTime(1n, new Temporal.TimeZone("UTC"));
```

Comment:
Please keep the amount of code inside the expect callback to the required minimum - only things actually taking part in the exception should be inside, the zdt should be created beforehand.
Reason being, if either of the ctors happen to throw the same error, we'd never notice it and the test would pass (!).